torch/onnx/symbolic_opset9.py

```diff
@@ -2692,53 +2692,57 @@ def arange(g, *args):
 
     def _get_arange_dtype(dtype):
         dtype = sym_help._maybe_get_const(dtype, "i")
-        if sym_help._is_value(dtype):
-            dtype = 4  # default to int64
         return dtype
 
-    if len(args) == 2:
-        # aten::arange(Scalar end, Tensor out)
-        end = sym_help._unsqueeze_helper(g, args[0], [0])
-        dtype = 4  # default to int64
-        arange_tensor = sym_help._squeeze_helper(g, nonzero(g, ones(g, end, dtype, None, None)), [1])
+    def _float_step_convert(range_tensor):
+        if sym_help._is_fp(range_tensor):
+            range_tensor = g.op("Cast", g.op("Ceil", range_tensor), to_i=sym_help.scalar_type_to_onnx[4])
```

Comment:
why is the cast to 4 required?

Reply:
This is for cases when steps are floats, their ceils are also floats -> hence range tensor is float as well.
Now this range tensor is called by ones() and subsequently as ConstantOfShape()'s sizes paramater, which raises an error if this value is not an int64, hence the cast node.